### PR TITLE
Add LS completion support for channel based worker interactions

### PR DESCRIPTION
**Pull request by rasika**
## Purpose
Completions support should be available when using `->`(worker trigger), `<-`(worker reply) symbols.

worker-receive
worker-send

This PR resolves https://github.com/ballerina-platform/ballerina-lang/issues/10717

### CODE REVIEW

**Comment by nadeeshaan** [comment]
@rasika this is awesome. I would like to point one improvement here, within the suggestions we do need to avoid the types as I think. we suggest the channel type as well. Shall we do that as well?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/FilterUtils.java [hunk 1]
```diff
@@ -57,9 +58,14 @@
...
         if (variable.getScopeEntry().symbol instanceof BEndpointVarSymbol
-                && delimiter.equals(UtilSymbolKeys.ACTION_INVOCATION_SYMBOL_KEY)
+                && delimiter.equals(UtilSymbolKeys.RIGHT_ARROW_SYMBOL_KEY)
                 && ((BEndpointVarSymbol) bVarSymbol).getClientFunction.type instanceof BInvokableType) {
+            //Handling action invocations ep -> ...
             resultList.addAll(getEndpointActions((BEndpointVarSymbol) variable.getScopeEntry().symbol));
+        } else if (delimiter.equals(UtilSymbolKeys.RIGHT_ARROW_SYMBOL_KEY)
+                || delimiter.equals(UtilSymbolKeys.LEFT_ARROW_SYMBOL_KEY)) {
+            // Handling worker-interactions eg. msg -> ... || msg <- ...
+            resultList.addAll(context.get(CompletionKeys.VISIBLE_SYMBOLS_KEY));
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/FilterUtils.java]: Let's do the symbol filtering and avoid the types and other unrelated symbols.

File: language-server/modules/langserver-core/src/test/resources/completion/worker/workerChannelReceive.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+    "position": {
+        "line": 7,
+        "character": 19
+    },
+    "source": "worker/source/workerChannelInteraction.bal",
+    "items": [
+        {
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/test/resources/completion/worker/workerChannelReceive.json]: Is this the only item we are getting in this particular scenario? If isn't shall we add all the returned items?

File: language-server/modules/langserver-core/src/test/resources/completion/worker/workerChannelSend.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+    "position": {
+        "line": 14,
+        "character": 16
+    },
+    "source": "worker/source/workerChannelInteraction.bal",
+    "items": [
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/test/resources/completion/worker/workerChannelSend.json]: Same as above

Comment (rasika) [reply language-server/modules/langserver-core/src/test/resources/completion/worker/workerChannelSend.json]: yes, only `chn` will be suffcient for this scenario IMO.

File: language-server/modules/langserver-core/src/test/resources/completion/worker/workerChannelReceive.json (hunk 2 above)
Comment (rasika) [reply language-server/modules/langserver-core/src/test/resources/completion/worker/workerChannelReceive.json]: yes, only `chn` will be sufficient for this scenario IMO.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/FilterUtils.java (hunk 1 above)
Comment (rasika) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/FilterUtils.java]: +1 removed unrelated symbols.

**Pull request merged**